lnwallet/channel.go

```diff
@@ -4098,12 +4089,26 @@ func (lc *LightningChannel) computeView(view *htlcView,
 	// channel constraints to the final commitment state. If any fee
 	// updates are found in the logs, the commitment fee rate should be
 	// changed, so we'll also set the feePerKw to this new value.
-	filteredHTLCView, uncommitted, err := lc.evaluateHTLCView(
-		view, &ourBalance, &theirBalance, whoseCommitChain,
+	filteredHTLCView, uncommitted, deltas, err := lc.evaluateHTLCView(
+		view, whoseCommitChain,
 	)
 	if err != nil {
 		return 0, 0, 0, nil, err
 	}
+
+	// Add the balance deltas to the balances we got from the commitment
+	// state.
+	if deltas.Local >= 0 {
+		ourBalance += lnwire.MilliSatoshi(deltas.Local)
+	} else {
```

Comment:
I think `ourBalance += lnwire.MilliSatoshi(deltas.Local)` is enough since `deltas.Local` can be negative.

Reply:
`deltas.Local` can be negative, but `MilliSatoshi` cannot, so the cast is likely to lead to undefined behavior.